legend-depot-pure-model-context/src/main/java/org/finos/legend/depot/server/pure/model/context/services/PureModelContextServiceImpl.java

```diff
@@ -67,15 +64,7 @@ public String getPureModelContextDataAsString(String groupId, String artifactId,
     @Override
     public PureModelContextData getPureModelContextData(String groupId, String artifactId, String versionId, String clientVersion, boolean versioned, boolean getDependencies)
     {
-        String version = versionId;
-        if (version.equals(VersionAlias.LATEST.getName()))
-        {
-            Optional<VersionId> project = this.projectsService.getLatestVersion(groupId, artifactId);
-            if (project.isPresent())
-            {
-                version = project.get().toVersionIdString();
-            }
-        }
+        String version = this.projectsService.resolveAliasesAndCheckVersionExists(groupId, artifactId, versionId);
```

Comment:
u dont need this code, the get entities already does the check ..

Reply:
yeah but `getPureModelContextData` also takes `version` and it doesn't handle alias.